pkg/storages/azure/folder.go

```diff
@@ -215,166 +278,38 @@ func (folder *Folder) ListFolder() (objects []storage.Object, subFolders []stora
-		// Any portion of the CanonicalizedResource string that is derived from
-		// the resource's URI should be encoded exactly as it is in the URI.
-		// -- https://msdn.microsoft.com/en-gb/library/azure/dd179428.aspx
-		cr.WriteString(u.EscapedPath())
-	} else {
-		// a slash is required to indicate the root path
-		cr.WriteString("/")
-	}
-
-	// params is a map[string][]string; param name is key; params values is []string
-	params, err := url.ParseQuery(u.RawQuery) // Returns URL decoded values
-	if err != nil {
-		return "", fmt.Errorf("failed to parse query params: %w", err)
-	}
-
-	if len(params) > 0 { // There is at least 1 query parameter
-		var paramNames []string // We use this to sort the parameter key names
-		for paramName := range params {
-			paramNames = append(paramNames, paramName) // paramNames must be lowercase
-		}
-		sort.Strings(paramNames)
-
-		for _, paramName := range paramNames {
-			paramValues := params[paramName]
-			sort.Strings(paramValues)
-
-			// Join the sorted key values separated by ','
-			// Then prepend "keyName:"; then add this string to the buffer
-			cr.WriteString("\n" + paramName + ":" + strings.Join(paramValues, ","))
-		}
-	}
-	return cr.String(), nil
-}
-// End from https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/zc_shared_policy_shared_key_credential.go
-
 func (folder *Folder) ReadObject(objectRelativePath string) (io.ReadCloser, error) {
 	path := storage.JoinPath(folder.path, objectRelativePath)
-	blobClient := folder.containerClient.NewBlobClient(path)
-	httpClient := &http.Client{ Timeout: folder.timeout }
-
-	req, err := http.NewRequest("GET", blobClient.URL(), nil)
+	blobClient, err := folder.containerClient.NewBlockBlobClient(path)
 	if err != nil {
-		return nil, NewFolderError(err, "Unable to download blob %s.", path)
-	}
-
-	if folder.credential != nil {
-		// Shared Key auth involves signing each request
-		if d := req.Header.Get("x-ms-data"); d == "" {
-			req.Header.Set("x-ms-date", time.Now().UTC().Format(http.TimeFormat))
-		}
-		stringToSign, stringErr := buildStringToSign(folder.credential, req)
-		if stringErr != nil {
-			return nil, NewFolderError(stringErr, "Unable to sign request to sign blob %s.", path)
-		}
-		signature, sigErr := folder.credential.ComputeHMACSHA256(stringToSign)
-		if sigErr != nil {
-			return nil, NewFolderError(sigErr, "Unable to sign request to sign blob %s.", path)
-		}
-		req.Header.Set("Authorization", strings.Join([]string{"SharedKey ", folder.credential.AccountName(), ":", signature}, ""))
+		return nil, NewFolderError(err, "Unable to init Azure Blob client")
 	}
 
-	resp, err := httpClient.Do(req)
-
+	get, err := blobClient.Download(context.Background(), nil)
 	if err != nil {
 		return nil, NewFolderError(err, "Unable to download blob %s.", path)
 	}
-
-	if resp.StatusCode < 200 || resp.StatusCode > 299 {
-		resp.Body.Close()
-		if resp.StatusCode == 404 {
-			return nil, storage.NewObjectNotFoundError(path)
```

Comment:
I don't see the reason why the 404 handling logic has been removed. I guess we still need to distinguish the 404 and return an appropriate `storage.NewObjectNotFoundError()` error.

Reply:
sorry, didn't realize 404s were specifically meaningful